workflow/workflow.go

```diff
@@ -571,11 +577,37 @@ func GetLastError(ctx Context) error {
 //
 // For supported operations on different server versions see [Visibility].
 //
+// Deprecated: use [UpsertTypedSearchAttributes] instead.
+//
 // [Visibility]: https://docs.temporal.io/visibility
 func UpsertSearchAttributes(ctx Context, attributes map[string]interface{}) error {
 	return internal.UpsertSearchAttributes(ctx, attributes)
 }
 
+// UpsertTypedSearchAttributes is used to add, update, or remove workflow search attributes. The search attributes can
+// be used in query of List/Scan/Count workflow APIs. The key and value type must be registered on temporal server side;
+// The value has to deterministic when replay; The value has to be Json serializable.
```

Comment:
`The value has to deterministic when replay`

It does?

Reply:
I just took this doc from `UpsertSearchAttributes`. But I just confirmed that we only check event type for upsert. So I will remove this statement from both Godocs on next commit (still going to wait for CLI publish to see if some CLI errors are solved).

Reply:
Updated